pkg/ddevapp/config.go

```diff
@@ -1066,8 +1063,8 @@ func (app *DdevApp) RenderComposeYAML() (string, error) {
 		XHProfMode:              app.GetXHProfMode(),
 
 		// Only use the extra_hosts technique for Linux and only if not WSL2 and not Colima
-		// If WSL2 we have to figure out other things, see GetHostDockerInternalIP()
-		UseHostDockerInternalExtraHosts: (runtime.GOOS == "linux" && !nodeps.IsWSL2() && !dockerutil.IsColima()) || (nodeps.IsWSL2() && globalconfig.DdevGlobalConfig.XdebugIDELocation == globalconfig.XdebugIDELocationWSL2),
+		// If WSL2 we have to figure out other things, see GetHostDockerInternal()
+		UseHostDockerInternalExtraHosts: hostDockerInternalExtraHost == "host-gateway",
```

Comment:
At some point we should pursue whether there would be a reliable consolidated way to do all this. But only if it would mean more reliability.